alma/lib/Traits/ObjectModelTrait.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * 2018-2024 Alma SAS.
+ *
+ * THE MIT LICENSE
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ * @author    Alma SAS <contact@getalma.eu>
+ * @copyright 2018-2024 Alma SAS
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ */
+
+namespace Alma\PrestaShop\Traits;
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+trait ObjectModelTrait
+{
+    /**
+     * Updates the current object in the database
+     *
+     * @param bool $null_values
+     *
+     * @return bool
+     *
+     * @throws \PrestaShopException
+     */
+    protected function updateWithFullyQualifiedNamespace($null_values = false)
+    {
+        // @hook actionObject*UpdateBefore
+        \Hook::exec('actionObjectUpdateBefore', ['object' => $this]);
+        \Hook::exec('actionObject' . $this->getFullyQualifiedName() . 'UpdateBefore', ['object' => $this]);
+
+        $this->clearCache();
+
+        // Automatically fill dates
+        if (array_key_exists('date_upd', $this)) {
+            $this->date_upd = date('Y-m-d H:i:s');
+            if (isset($this->update_fields) && is_array($this->update_fields) && count($this->update_fields)) {
+                $this->update_fields['date_upd'] = true;
+            }
+        }
+
+        // Automatically fill dates
+        if (array_key_exists('date_add', $this) && $this->date_add == null) {
```

Comment:
is $this an array or an object?

Reply:
This function is copied from PrestaShop code to fix the issue in old PrestaShop version
https://github.com/PrestaShop/PrestaShop/blob/b48f3c284c198586804322055f4fa1bb292c17b8/classes/ObjectModel.php#L650
I can update this function from their last code

Reply:
This function is executed until the merchant with Prestashop 1.7.1.0. It's why I used the code from this version